dual_net.py

```diff
@@ -81,6 +83,8 @@ def bootstrap(self):
             tf.train.Saver().save(sess, self.save_file)
 
     def train(self, tf_records, init_from=None, logdir=None, num_steps=None):
+        def should_log(i):
```

Comment:
How about making this a top-level function

log_sample(i):

Or do you need all the closure vars?

Reply:
the closure var is used.  This way is both DRY and readable.